pattern/pattern.c

```diff
@@ -180,6 +256,21 @@ void pattern_render(struct pattern * pattern, GLuint input_tex) {
         loc = glGetUniformLocationARB(pattern->shader[i], "iChannel");
         glUniform1ivARB(loc, pattern->n_shaders, pattern->uni_tex);
 
+        if (pattern->frames) {
+            int texture_index = pattern->n_shaders + 1;
+            glActiveTexture(GL_TEXTURE0 + texture_index);
+            glBindTexture(GL_TEXTURE_2D, pattern->frames[pattern->current_frame]);
+
+            loc = glGetUniformLocationARB(pattern->shader[i], "iImage");
+            glUniform1iARB(loc, texture_index);
+
+            // TODO: Should do something interesting off of beat_frac and beat_index
+            // to make the animation go faster/slower depending on the music
+            if (pattern->last_ms / RADIANCE_PATTERN_GIF_SPEED != time_master.wall_ms / RADIANCE_PATTERN_GIF_SPEED) {
```

Comment:
So if I'm reading this right (apologies if not), the first time we render we always show frame 0, but then the next frame jumps somewhere random in the animation (as we take the wall time in absolute ms / PATTERN_GIF_SPEED).  Also if we stop rendering this pattern for a while (by switching decks? I've never actually used this program lol) we'll again render one frame from the last position and then jump.

Consider moving the frame selection before the texture bind, and setting last_ms to wall_ms if last_ms is either zero or perhaps far from the current time.  Then move the index by (wall_ms - last_ms) / GIF_SPEED to animate relative to the first render.